src/consumer/mod.rs

```diff
@@ -593,6 +605,48 @@ mod tests {
             assert_eq!(consumer.messages_received(), 4);
             assert!(consumer.last_message_received().is_some());
         }
+
+        let stats = consumer_1.get_stats().await.unwrap();
+        assert_eq!(stats.len(), 2);
+        for stat in stats {
+            assert_eq!(stat.consumer_name().to_string(), "consumer_1");
+        }
+
+        let data5 = TestData {
+            topic: "c".to_owned(),
+            msg: 5,
+        };
+        let data6 = TestData {
+            topic: "c".to_owned(),
+            msg: 6,
+        };
+        try_join_all(vec![
+            client.send(&topic1, &data5),
+            client.send(&topic1, &data6),
+        ])
+        .await
+        .unwrap();
+        let mut latest_msg = None;
+        for i in 0..2 {
+            let msg = recv_within(&mut consumer_1, DEFAULT_RECV_TIMEOUT).await;
+            println!("consumer_1 receive {}: {:?}", i, msg);
+            if i == 1 {
+                latest_msg = Some(msg.unwrap());
+            }
+        }
+
+        consumer_1
+            .cumulative_ack(&latest_msg.unwrap())
+            .await
+            .unwrap();
+        let msg = recv_within(&mut consumer_1, DEFAULT_RECV_TIMEOUT).await;
```

Comment:
It would be better to reduce the timeout here to reduce test time.

A better test pattern to verify ack works is to send a new message and recreate the consumer to consume it.

Reply:
When a consumer is created and then a new message is sent, the consumer will by default use the [latest message](https://github.com/streamnative/pulsar-rs/blob/f2cbcf1e0fe55508476f4ca48f1e26581e66ad75/src/consumer/initial_position.rs#L8) consumption information, which does not verify that previous messages have been acknowledged.